Scrapping_part_3/scrapping_part3/scrapping_part3/middlewares.py

```diff
@@ -0,0 +1,46 @@
+from stem import Signal
+from stem.control import Controller
+from scrapy.downloadermiddlewares.httpproxy import HttpProxyMiddleware
+from stem.util.log import get_logger
+
+from .settings import REQUESTS_PER_SAME_TOR_IDENTITY
+
+
+# Number of requests sent during the current tor identity
+NUM_SENT_REQUESTS = 0
+
+
+def new_tor_identity():
+    with Controller.from_port(port=9051) as controller:
+        # authenticate with the tor password
+        controller.authenticate(password='027D1370DD5D27AA602286D179466905FD6A15778772556AFBAE310CCD')
```

Comment:
It's a very bad practice to store password in a VCS, once you've added, you've never remove it (it will remain in history).